public/app/features/variables/adhoc/urlParser.ts

```diff
@@ -17,11 +17,19 @@ export const toFilters = (value: UrlQueryValue): AdHocVariableFilter[] => {
 };
 
 function escapeDelimiter(value: string | undefined): string {
-  return value?.replace(/\|/g, '__gfp__') ?? '';
+  if (value === null || value === undefined) {
+    return '';
+  }
+
+  return /\|/g[Symbol.replace](value, '__gfp__');
```

Comment:
@hugohaggmark interesting, can you say few words how does this help?

Reply:
That method handles numbers, strings, boolean etc but `null` and `undefined` are just converted to `"null"` and `"undefined"` so I still need the if statement unfortunately.

Reply:
OK, I see, thanks for the explanation <3

Reply:
It's like string.replace method but on a regex instance. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/@@replace#direct_call